osu.Game/Screens/OnlinePlay/Multiplayer/Match/Playlist/MultiplayerHistoryList.cs

```diff
@@ -15,16 +20,82 @@ namespace osu.Game.Screens.OnlinePlay.Multiplayer.Match.Playlist
     /// </summary>
     public partial class MultiplayerHistoryList : DrawableRoomPlaylist
     {
+        public new Bindable<PlaylistItem?> SelectedItem => throw new NotSupportedException();
+
+        [Resolved]
+        private MultiplayerClient client { get; set; } = null!;
+
+        private bool firstPopulation = true;
+
         public MultiplayerHistoryList()
         {
             ShowItemOwners = true;
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            client.RoomUpdated += onRoomUpdated;
+            client.ItemAdded += onItemAdded;
+            client.ItemRemoved += onItemRemoved;
+            client.ItemChanged += onItemChanged;
+
+            onRoomUpdated();
+        }
+
+        private void onRoomUpdated() => Scheduler.AddOnce(() =>
+        {
+            if (client.Room == null)
+            {
+                Items.Clear();
+                firstPopulation = true;
+            }
+            else if (firstPopulation)
+            {
+                foreach (var item in client.Room.Playlist)
+                    onItemAdded(item);
+                firstPopulation = false;
+            }
```

Comment:
Can `MultiplayerClient` not fire `ItemAdded` on initial join or something to avoid playing these `firstPopulation` games here? Because I see that at least 2 components are playing the same boolean flag games and at that point it feels rather icky.